### PR TITLE
Tensorboard visualization tool for MLP, XGBoost and CatBoost

### PR DESCRIPTION
**Pull request by igor17400**
## Description
TensorBoard provides the visualization and tooling needed for machine learning experimentation such as tracking and visualizing metrics such as loss and accuracy, visualizing the model graph (ops and layers) and much more.

## Motivation and Context
This tool can bring features that will help to analyze the loss of the model being applied as well as finacial metrics in order to understand how well one model is being trained and validated.

## How Has This Been Tested?
- [x] Pass the test by running: `pytest qlib/tests/test_all_pipeline.py` under upper directory of `qlib`.
- [x] If you are adding a new feature, test on your own test scripts.
In order to test such feature just execute any of the following `yml` files:
-  `workflow_config_xgboost_Alpha360.yaml`,
- `workflow_config_catboost_Alpha360.yaml` and 
- `workflow_config_mlp_Alpha360.yaml`

## Screenshots of Test Results (if appropriate):
1. Pipeline test:
2. Your own tests:

RMSE for each step on training and validation iterations for CatBoost:

Screen Shot 2022-05-10 at 08 46 07

RMSE for each step on training and validation iterations for XGBoost:

Screen Shot 2022-05-10 at 08 47 51

MSE for each step on training and validation iterations for MLP as well as IC (Pearson correlation coefficient):

Screen Shot 2022-05-10 at 08 47 29

Screen Shot 2022-05-10 at 08 47 39

## Types of changes
- [ ] Fix bugs
- [x] Add new feature
- [ ] Update documentation

### CODE REVIEW

File: qlib/contrib/model/catboost_model.py [hunk 1]
```diff
@@ -12,18 +15,32 @@
 from ...data.dataset.handler import DataHandlerLP
 from ...model.interpret.base import FeatureInt
 from ...data.dataset.weight import Reweighter
+from ...log import get_module_logger
+
+working_dir = os.getcwd()
+target_dir = os.path.join(os.path.relpath(working_dir), "tensor_board")
+target_dir += "/runs/catboost/"
+sys.path.insert(0, target_dir)
```
Comment (you-n-g) [inline qlib/contrib/model/catboost_model.py:23]: Why do you have to change `sys.path`?
Do you want to import any Python module from this path?

File: qlib/contrib/model/catboost_model.py [hunk 2]
```diff
@@ -12,18 +15,32 @@
 from ...data.dataset.handler import DataHandlerLP
 from ...model.interpret.base import FeatureInt
 from ...data.dataset.weight import Reweighter
+from ...log import get_module_logger
+
+working_dir = os.getcwd()
+target_dir = os.path.join(os.path.relpath(working_dir), "tensor_board")
```
Comment (you-n-g) [inline qlib/contrib/model/catboost_model.py:21]: I think we can avoid using it as a global variable.
Maybe local variable will be enough and better.

File: qlib/contrib/model/catboost_model.py [hunk 3]
```diff
@@ -12,18 +15,32 @@
...
+target_dir += "/runs/catboost/"
+sys.path.insert(0, target_dir)
 
 
 class CatBoostModel(Model, FeatureInt):
     """CatBoost Model"""
 
-    def __init__(self, loss="RMSE", **kwargs):
+    def __init__(self, loss="RMSE", tensorboard=False, tensorboard_name="", **kwargs):
```
Comment (you-n-g) [inline qlib/contrib/model/catboost_model.py:29]: Do you think merging `tensorboard=False, tensorboard_name=""` to a single one will make the code more concise?
For example
````Python
tensorboard_name: Optional[str] = None
````
`None` indicates that tensorboard is not enabled.

File: qlib/contrib/model/catboost_model.py [hunk 4]
```diff
@@ -12,18 +15,32 @@
 from ...data.dataset.handler import DataHandlerLP
 from ...model.interpret.base import FeatureInt
 from ...data.dataset.weight import Reweighter
+from ...log import get_module_logger
+
+working_dir = os.getcwd()
+target_dir = os.path.join(os.path.relpath(working_dir), "tensor_board")
+target_dir += "/runs/catboost/"
```
Comment (you-n-g) [inline qlib/contrib/model/catboost_model.py:22]: using `pathlib`  or  `os.path.join` to join path looks more standard.

File: qlib/contrib/model/xgboost.py [hunk 5]
```diff
@@ -10,15 +12,24 @@
 from ...data.dataset.handler import DataHandlerLP
 from ...model.interpret.base import FeatureInt
 from ...data.dataset.weight import Reweighter
+from ...log import get_module_logger
+
+from tensorboardX import SummaryWriter
```
Comment (you-n-g) [inline qlib/contrib/model/xgboost.py:17]: If you want tensorboard to be a required feature,  please add it into the `requriements.txt` of Catboost and xgboost.

File: tensor_board/README.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
+# Tensorboard
```
Comment (you-n-g) [inline tensor_board/README.md:1]: Tensorboard is not currently an essential part of QLib's framework.
I think we can put it into `https://github.com/microsoft/qlib/tree/main/examples/benchmarks` before we support it in the framework level.

For example,  the path could be `examples/benchmarks/docs/tensor_board.md`

**Comment by you-n-g** [comment]
Sorry for the late response @igor17400  .
I added some comments just now.

Besides,
please fix the pylint error.
https://github.com/microsoft/qlib/runs/6369431237?check_suite_focus=true#step:7:34

**Comment by igor17400** [comment]
Thanks @you-n-g for the replies!
I'll work on the suggestions and commit the updated code

**Comment by you-n-g** [comment]
The main branch has changed a lot (especially the CI).
Please merge the main branch.
Feel free to contact us if you encountered any problem when fixing the issue.
